extensions/eda/plugins/event_source/file_watch.py

```diff
@@ -41,47 +41,55 @@ def __init__(self: "Handler", **kwargs: Any) -> None:
             RegexMatchingEventHandler.__init__(self, **kwargs)
 
         def on_created(self: "Handler", event: FileSystemEvent) -> None:
-            loop.call_soon_threadsafe(
-                queue.put_nowait,
-                {
-                    "change": "created",
-                    "src_path": event.src_path,
-                    "type": event.__class__.__name__,
-                    "root_path": root_path,
-                },
+            asyncio.run_coroutine_threadsafe(
```

Comment:
@Alex-Izquierdo Should this be using the await put call like the other source [plugins](https://github.com/ansible/event-driven-ansible/blob/586ea78462ebe506b788224d346011bea3c8f3b6/extensions/eda/plugins/event_source/aws_sqs_queue.py#L82).

Reply:
@mkanoor if you look at the code, watchdog is sync code that is executed with a [threadpool](https://github.com/ansible/event-driven-ansible/blob/fd3e8e1695acaedac9a5c8a88fbdf8604ab02e60/extensions/eda/plugins/event_source/file_watch.py#L111) this is not a pure asyncio plugin.